src/test/java/spring/boot/springapp/crud/UserCrudTest.java

```diff
@@ -0,0 +1,55 @@
+package spring.boot.springapp.crud;
+
+import java.time.LocalDate;
+import java.util.Set;
+import org.bson.types.ObjectId;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.data.mongo.AutoConfigureDataMongo;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import spring.boot.springapp.model.Role;
+import spring.boot.springapp.model.User;
+import spring.boot.springapp.service.UserService;
+
+@AutoConfigureDataMongo
+@TestPropertySource(locations = "classpath:test.properties")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@ExtendWith(SpringExtension.class)
+public class UserCrudTest {
+    private static final Role USER = new Role();
+    private static final Role ADMIN = new Role();
+    private final UserService userService;
+
+    @Autowired
+    public UserCrudTest(UserService userService) {
+        this.userService = userService;
+    }
+
+    @BeforeAll
+    public static void start(@Autowired MongoTemplate mongoTemplate) {
+        mongoTemplate.getDb().drop();
+        USER.setRole(Role.RoleName.USER);
+        USER.setId(ObjectId.get());
+        ADMIN.setRole(Role.RoleName.ADMIN);
+        ADMIN.setId(ObjectId.get());
+    }
+
+    @Test
+    public void saveUser_Ok() {
```

Comment:
@Artificial-Friend Let's add a test case for some specific methods. For example, `find by phone number`. Because it makes not much sense to test the default `save` method.

Reply:
I understand, It just wasn't a requirement so it's only a start point.